alien4cloud/alien4cloud
# Fix missing upload max size when deactivate spring multipart file upload

# Description

The commit https://github.com/alien4cloud/alien4cloud/commit/010e2ecc58056430cc8fd0c9a86da38d6fea57ea has deactivated the **spring multipart file upload** and replaced it by the **apache fileupload**, but also disabled the config `upload.max_archive_size`.

Without specifying the limit size, the apache fileupload has no limit by default: 
https://commons.apache.org/proper/commons-fileupload/javadocs/api-release/index.html

This has a potential risk in security.

# Solution

This PR:
* Add config `upload.max_archive_size` to limit upload file of csar and artifact. Defaults to 10Mb.
* Add config `upload.max_image_size` to limit upload file of an image. Defaults to 100Kb.

# How to test

1. Set max_archive_size in `alien4cloud-config.yml` with a small value:

```
upload:
  max_archive_size: 1024
```

2. Upload a csar bigger than 1Kb
3. See error

![error](https://i.imgur.com/OZ1NJti.png)
